---
title: '12\.Phylogenetic Diversity - Communities'
author: "Student Name; Z620: Quantitative Biodiversity, Indiana University"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
geometry: margin=2.54cm
---

## OVERVIEW
Complementing taxonomic measures of $\alpha$- and $\beta$-diversity with evolutionary information yields insight into a broad range of biodiversity issues including conservation, biogeography, and community assembly. 
In this worksheet, you will be introduced to some commonly used methods in phylogenetic community ecology.

After completing this assignment you will know how to:

1.  incorporate an evolutionary perspective into your understanding of community ecology
2.  quantify and interpret phylogenetic $\alpha$- and $\beta$-diversity
3.  evaluate the contribution of phylogeny to spatial patterns of biodiversity

## Directions:
1. In the Markdown version of this document in your cloned repo, change "Student Name" on line 3 (above) with your name.
2. Complete as much of the worksheet as possible during class.
3. Use the handout as a guide; it contains a more complete description of data sets along with examples of proper scripting needed to carry out the exercises.
4. Answer questions in the  worksheet.
Space for your answers is provided in this document and is indicated by the ">" character.
If you need a second paragraph be sure to start the first line with ">".
You should notice that the answer is highlighted in green by RStudio (color may vary if you changed the editor theme). 
5. Before you leave the classroom today, it is *imperative* that you **push** this file to your GitHub repo, at whatever stage you are. This will enable you to pull your work onto your own computer.
6. When you have completed the worksheet, **Knit** the text and code into a single PDF file by pressing the `Knit` button in the RStudio scripting panel.
This will save the PDF output in your '12.PhyloCom' folder.
7. After Knitting, please submit the worksheet by making a **push** to your GitHub repo and then create a **pull request** via GitHub.
Your pull request should include this file *12.PhyloCom_Worksheet.Rmd* and the PDF output of `Knitr` (*12.PhyloCom_Worksheet.pdf*).

The completed exercise is due on **Monday, May 10^th^, 2021 before 09:00 AM**.

## 1) SETUP

Typically, the first thing you will do in either an R script or an RMarkdown file is setup your environment. 
This includes things such as setting the working directory and loading any packages that you will need.

In the R code chunk below, provide the code to:  
1. clear your R environment,  
2. print your current working directory,  
3. set your working directory to your `/12.PhyloCom` folder,  
4. load all of the required R packages (be sure to install if needed), and  
5. load the required R source file.

```{r}
rm(list = ls())
setwd("~/Classwork/Current/QB/")
```

## 2) DESCRIPTION OF DATA

### need to discuss data set from spatial ecology!
 
In 2013 we sampled > 50 forested ponds in Brown County State Park, Yellowood State Park, and Hoosier National Forest in southern Indiana. 
In addition to measuring a suite of geographic and environmental variables, we characterized the diversity of bacteria in the ponds using molecular-based approaches.
Specifically, we amplified the 16S rRNA gene (i.e., the DNA sequence) and 16S rRNA transcripts (i.e., the RNA transcript of the gene) of bacteria.
We used a program called `mothur` to quality-trim our data set and assign sequences to operational taxonomic units (OTUs), which resulted in a site-by-OTU matrix.  
In this module we will focus on taxa that were present (i.e., DNA), but there will be a few steps where we need to parse out the transcript (i.e., RNA) samples.
See the handout for a further description of this week's dataset.

## 3) LOAD THE DATA

In the R code chunk below, do the following:  
1. load the environmental data for the Brown County ponds (*20130801_PondDataMod.csv*),  
2. load the site-by-species matrix using the `read.otu()` function,  
3. subset the data to include only DNA-based identifications of bacteria,  
4. rename the sites by removing extra characters,  
5. remove unnecessary OTUs in the site-by-species, and   
6. load the taxonomic data using the `read.tax()` function from the source-code file.

```{r}
for (p in c('picante', 'ape', 'seqinr', 'vegan', 'fossil', 'reshape', 'simba')) {
  require(p, character.only=TRUE)
}
source("./bin/MothurTools.R")
```

Next, in the R code chunk below, do the following:  
1. load the FASTA alignment for the bacterial operational taxonomic units (OTUs),  
2. rename the OTUs by removing everything before the tab (\\t) and after the bar (|),  
3. import the *Methanosarcina* outgroup FASTA file,  
4. convert both FASTA files into the DNAbin format and combine using `rbind()`,  
5. visualize the sequence alignment,  
6. using the alignment (with outgroup), pick a DNA substitution model, and create a phylogenetic distance matrix,  
7. using the distance matrix above, make a neighbor joining tree,  
8. remove any tips (OTUs) that are not in the community data set,  
9. plot the rooted tree.

```{r}
env <- read.table("data/20130801_PondDataMod.csv", sep = ',', header = TRUE)
env <- na.omit(env)

# load a select data
comm <- read.otu(shared = "./data/INPonds.final.rdp.shared", cutoff = "1")
comm <- comm[grep("*-DNA", rownames(comm)), ]
rownames(comm) <- gsub("\\-DNA", "", rownames(comm))
rownames(comm) <- gsub("\\_", "", rownames(comm))

# remove missing sites and zeros
comm <- comm[rownames(comm) %in% env$Sample_ID,]
comm <- comm[, colSums(comm) > 0]
tax <- read.tax(taxonomy = "./data/INPonds.final.rdp.1.cons.taxonomy")

# import alignment file
ponds.cons <- read.alignment(file = "./data/INPonds.final.rdp.1.rep.fasta",
                             format = "fasta")
ponds.cons$nam <- gsub("\\|.*$", "", gsub("^.*?\t", "", ponds.cons$nam))

# import outgroup sequence
outgroup <- read.alignment(file = "./data/methanosarcina.fasta", format = "fasta")
DNAbin <- rbind(as.DNAbin(outgroup), as.DNAbin(ponds.cons))
image.DNAbin(DNAbin, shot.labels = T, cex.lab = 0.05, las = 1)

# distence matrix
seq.dist.jc <- dist.dna(DNAbin, model = "JC", pairwise.deletion = FALSE)
#tree
phy.all <- bionj(seq.dist.jc)
phy <- drop.tip(phy.all, phy.all$tip.label[!phy.all$tip.label %in% c(colnames(comm), "Methanosarcina")])

# pull out the outgroup w.r.t the tree and root it
outgroup <- match("Methanosarcina", phy$tip.label)
phy <- root(phy, outgroup, resolve.root = TRUE)
par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(phy, main = "Neighbor Joining Tree", "phylogram", show.tip.label = FALSE,
           use.edge.length = FALSE, direction = "right", cex = 0.6, label.offset = 1)

```

## 4) PHYLOGENETIC ALPHA DIVERSITY

### A.  Faith's Phylogenetic Diversity (PD)
In the R code chunk below, do the following:  
1. calculate Faith's D using the `pd()` function.

```{r}
require(ggplot2)
faith_pd <- pd(comm, phy, include.root = FALSE)
line <- lm(log(PD) ~ log(SR), faith_pd)
faith_pd
anova(line)
```

In the R code chunk below, do the following:  
1. plot species richness (S) versus phylogenetic diversity (PD),  
2. add the trend line, and  
3. calculate the scaling exponent. 

```{r}
print(line)
g <- ggplot(faith_pd, aes(x = log(SR), y = log(PD))) +
  geom_abline(slope = line$coefficients[2], intercept = line$coefficients[1], color = 'blue') +
  geom_point() +
  annotate("text", x = 6.15, y = 3.8, label = paste("Scaling Exponent = ", round(line$coefficients[2], 2), sep = "")) +
  xlab("ln(S)") +
  ylab("ln(PD)") +
  labs(title = "Phylodiversity (PD) vs. Taxonomic Richness (S)") +
  theme_classic()
g
```

***Question 1***: Answer the following questions about the PD-S pattern.    
a.  Based on how PD is calculated, why should this metric be related to taxonmic richness? 
b.  Describe the relationship between taxonomic richness and phylodiversity.
c.  When would you expect these two estimates of diversity to deviate from one another? 
d.  Interpret the significance of the scaling PD-S scaling exponent.

> ***Answer 1a***:  PD requires different taxa to provide diversity but not all additional taxa will provide the same amount of phylodiversity. Initial samples are more likely to represent taxa less related to those already sampled thus we expect a saturating relationship for a given site.
> ***Answer 1b***:  We see a sublinear power law relationship meaning that each additional taxon sampled provides less overall phylodiversity.
> ***Answer 1c***:  These measures should diverge anytime there are a lot of taxa to sample when PD will saturate.
> ***Answer 1d***:  The scaling exponent of 0.75 is highly significant giving us confidence that it is accurate.

**i. Randomizations and Null Models**  

In the R code chunk below, do the following:  
1. estimate the standardized effect size of PD using the `richness` randomization method.  

```{r}
rich_ses <- ses.pd(comm[1:2,], phy, null.model = "richness", runs = 25,
                 include.root = FALSE)
labl_ses <- ses.pd(comm[1:2,], phy, null.model = "taxa.labels", runs = 25,
                 include.root = FALSE)
ppool_ses <- ses.pd(comm[1:2,], phy, null.model = "phylogeny.pool", runs = 25,
                 include.root = FALSE)

rich_ses
labl_ses
ppool_ses
```

***Question 2***: Using `help()` and the table above, run the `ses.pd()` function using two other null models and answer the following questions:

a.  What are the null and alternative hypotheses you are testing via randomization when calculating `ses.pd`?
b.  How did your choice of null model influence your observed ses.pd values?  Explain why this choice affected or did not affect the output. 

> ***Answer 2a***:  I am testing three null hypotheses: that the identity of the species in the local site do not effect Faith's PD (`richness`); that the position of taxa on the phylogeny in our sites has no effect on the value of Faith's PD (`taxa.labels`); and that the local site is not different than the region in terms of Faith's PD (`phylogeny.pool`). The alternative hypotheses are that there are effects on Faith's PD on these features.
> ***Answer 2b***:  Under these null models, with the stochastic run I am looking at now, the interpretation is the same for all three. This is that at the second included pond, we almost see an effect and in the first we definitely don't. In the second pond PD is about 1 standard deviation off of the mean under each of the null models which is not a very powerful effect. This kind of makes sense in that each of these null models only allow changes to the site species identities but don't constrain anything about the species themselves. [I dont quite understand some of the null models so I've chosen ones that I do understand. In particular I dont get what maintaining species occurance frequency means? Is it the number of sites in which a species occurs?? `independentswap` is a null model I talked about in my quals today if so!]

### B.  Phylogenetic Dispersion Within a Sample
Another way to assess phylogenetic $\alpha$-diversity is to look at dispersion within a sample. 

**i. Phylogenetic Resemblance Matrix**

In the R code chunk below, do the following:  
1. calculate the phylogenetic resemblance matrix for taxa in the Indiana ponds data set.

```{r}
phydist <- cophenetic.phylo(phy)
```

**ii.  Net Relatedness Index (NRI)**

In the R code chunk below, do the following:  
1. Calculate the NRI for each site in the Indiana ponds data set. 

```{r}
ses_mpd <- ses.mpd(comm, phydist, null.model = "taxa.labels", 
                   abundance.weighted = TRUE, runs = 25)

NRI <- as.matrix(-1*((ses_mpd[,2] - ses_mpd[,3]) / ses_mpd[,4]))
rownames(NRI) <- row.names(ses_mpd)
colnames(NRI) <- "NRI"
NRI
```

**iii.  Nearest Taxon Index (NTI)**

In the R code chunk below, do the following:
1. Calculate the NTI for each site in the Indiana ponds data set. 

```{r}
ses_mntd <- ses.mntd(comm, phydist, null.model = "taxa.labels", 
                   abundance.weighted = TRUE, runs = 25)

NTI <- as.matrix(-1*((ses_mntd[,2] - ses_mntd[,3]) / ses_mntd[,4]))
rownames(NTI) <- row.names(ses_mntd)
colnames(NTI) <- "NTI"
NTI
```

***Question 3***: 

a. In your own words describe what you are doing when you calculate the NRI. 
b. In your own words describe what you are doing when you calculate the NTI. 
c. Interpret the NRI and NTI values you observed for this dataset. 
d. In the NRI and NTI examples above, the arguments "abundance.weighted = FALSE" means that the indices were calculated using presence-absence data.
Modify and rerun the code so that NRI and NTI are calculated using abundance data. 
How does this affect the interpretation of NRI and NTI?

> ***Answer 3a***:  NRI is the offset of the average branch lengths within a site from a null distribution of the same measure but expressed in terms of null distribution standard deviations. In other words it is the negative Z-score for the average branch length for the focal site against a null distribution.
> ***Answer 3b***:  NTI is the same idea---negative Z-score---except that the measure is the phylogenetic distance from each taxon to its nearest neighbor averaged over taxa.
> ***Answer 3c***:  Both of these measures, particularly NRI, indicate that sites are overdispersed with respect to the phylogeny. This is shown in the negative values and indicates that present taxa are less related than expected by chance.
> ***Answer 3d***:  This change completely changes the results and instead reports that most sites are underdispersed. Maybe a small number of closely related and highly abundant species are present in each of these sites?

## 5) PHYLOGENETIC BETA DIVERSITY 

### A. Phylogenetically Based Community Resemblance Matrix
In the R code chunk below, do the following:  
1. calculate the phylogenetically based community resemblance matrix using Mean Pair Distance, and  
2. calculate the phylogenetically based community resemblance matrix using UniFrac distance.

```{r}
dist_mp <- comdist(comm, phydist)
dist_uni <- unifrac(comm, phy)
```

In the R code chunk below, do the following:  
1. plot Mean Pair Distance versus UniFrac distance and compare.

```{r}
par(mar = c(5, 5, 2, 1) + 0.1)
plot(dist_mp, dist_uni, pch = 20, col = "red", las = 1, asp = 1, 
     xlim = c(0.15, 0.5), ylim = c(0.15, 0.5), xlab = "Mean Pair Distance",
     ylab = "UniFrac Distance")
abline(b = 1, a = 0, lty = 2)
text(0.5, 0.47, "1:1")
```

***Question 4***: 

a. In your own words describe Mean Pair Distance, UniFrac distance, and the difference between them.
b. Using the plot above, describe the relationship between Mean Pair Distance and UniFrac distance. 
Note: we are calculating unweighted phylogenetic distances (similar to incidence based measures).
That means that we are not taking into account the abundance of each taxon in each site.
c. Why might MPD show less variation than UniFrac?

> ***Answer 4a***:  Mean pair distance is what the name says: the mean pairwise phylogenetic distance between taxa. UniFrac is the sum of branch lengths for branches that are not in the intersection of both sites' set of branches normalized by the sum of the total branch lengths. UniFrac is not based on pairwise measures.
> ***Answer 4b***:  These measures provide very different results. Mean pairwise distance shows much less variability than UniFrac.
> ***Answer 4c***:  Thos is probably because of the fac that mean pairwise difference is always iusing the same tree without weights according to abundance so the differences between sites may just be short branches near te tips.


### B. Visualizing Phylogenetic Beta-Diversity
Now that we have our phylogenetically based community resemblance matrix, we can visualize phylogenetic diversity among samples using the same techniques that we used in the $\beta$-diversity module from earlier in the course.

In the R code chunk below, do the following:  
1. perform a PCoA based on the UniFrac distances, and  
2. calculate the explained variation for the first three PCoA axes.

```{r}
# calculate PCoA, explain variance, plot it
pond_pcoa <- cmdscale(dist_uni, eig = TRUE, k = 3)
expvar <- round(sapply(pond_pcoa$eig, function(x) x / sum(pond_pcoa$eig)), 3) * 100
plot_pcoa <- tibble::as_tibble(pond_pcoa$points)
```

Now that we have calculated our PCoA, we can plot the results.

In the R code chunk below, do the following:  
1. plot the PCoA results using either the R base package or the `ggplot` package,  
2. include the appropriate axes,   
3. add and label the points, and    
4. customize the plot. 

```{r}
g <- ggplot(plot_pcoa, aes(x = V1, y = V2)) +
  geom_vline(xintercept = 0, linetype=3) +
  geom_hline(yintercept = 0, linetype=3) +
  geom_point(size = 3, color="grey50") +
  geom_text(label = row.names(pond_pcoa$points)) +
  xlab(paste("PCoA 1 (", expvar[1], "%)", sep = "")) +
  ylab(paste("PCoA 2 (", expvar[2], "%)", sep = "")) +
  theme_classic()
g
```

In the following R code chunk:
1. perform another PCoA on taxonomic data using an appropriate measure of dissimilarity, and
2. calculate the explained variation on the first three PCoA axes. 

```{r}
# calculate PCoA, explain variance, plot it
comm_pcoa <- cmdscale(vegdist(comm, binary = TRUE), eig = TRUE, k = 3)
expvar <- round(sapply(comm_pcoa$eig, function(x) x / sum(comm_pcoa$eig)), 3) * 100
print(expvar)
```

***Question 5***: Using a combination of visualization tools and percent variation explained, how does the phylogenetically based ordination compare or contrast with the taxonomic ordination? 
What does this tell you about the importance of phylogenetic information in this system?

> ***Answer 5***:  We actually see less variance explained in the first few axes of the phylogenetically informed diatance matrix actually explains less variance than Sorenson dissimilarity. Maybe phylogenetic information is not as important as abundance information in this system.

### C. Hypothesis Testing

**i. Categorical Approach**

In the R code chunk below, do the following:  
1. test the hypothesis that watershed has an effect on the phylogenetic diversity of bacterial communities.

```{r}
watershed <- env$Location
adonis(dist_uni ~ watershed, permutations = 999)
```

**ii. Continuous Approach**

In the R code chunk below, do the following:
1. from the environmental data matrix, subset the variables related to physical and chemical properties of the ponds, and  
2. calculate environmental distance between ponds based on the Euclidean distance between sites in the environmental data matrix (after transforming and centering using `scale()`).

```{r}
envs <- env[, 5:19]
envs <- envs[, -which(names(envs) %in% c("TDS", "Salinity", "Cal_Volume"))]
env_dist <- vegdist(scale(envs), method = "euclid")
```

In the R code chunk below, do the following:  
1. conduct a Mantel test to evaluate whether or not UniFrac distance is correlated with environmental variation.

```{r}
mantel(dist_uni, env_dist)
```

Last, conduct a distance-based Redundancy Analysis (dbRDA). 

In the R code chunk below, do the following:  
1. conduct a dbRDA to test the hypothesis that environmental variation effects the phylogenetic diversity of bacterial communities,  
2. use a permutation test to determine significance, and
3. plot the dbRDA results

```{r}
require(dplyr)
# hypothesis test
ponds_dbrda <- vegan::dbrda(dist_uni ~ ., as.data.frame(scale(envs)))
#anova(ponds_dbrda, by = "axis")
#ponds_fit <- envfit(ponds_dbrda, envs, perm = 999)
#ponds_fit

# visualize
expvar <- round(sapply(ponds_dbrda$CCA$eig, function(x) x / sum(c(ponds_dbrda$CCA$eig, ponds_dbrda$CA$eig))), 3) * 100
plot_dbrda <- tibble::as_tibble(scores(ponds_dbrda)$sites)
env_vecs <- tibble::as_tibble(scores(ponds_dbrda, display = "bp")) %>% mutate(name = names(envs))
plot_dbrda
g <- ggplot() +
  geom_hline(yintercept = 0, linetype = 3) +
  geom_vline(xintercept = 0, linetype = 3) +
  geom_point(data = plot_dbrda, aes(x = dbRDA1, y = dbRDA2)) +
  geom_segment(data = env_vecs, aes(x = 0, y = 0, xend = dbRDA1, yend = dbRDA2), color='blue', arrow = arrow(length = unit(0.03, "npc"))) +
  geom_text(data = env_vecs, aes(x = dbRDA1, y = dbRDA2 + 0.1, label = name)) +
  xlab(paste("dbRDA 1 (", expvar[1], "%)", sep = "")) +
  ylab(paste("dbRDA 2 (", expvar[2], "%)", sep = "")) +
  theme_classic()
g
```

***Question 6***: Based on the multivariate procedures conducted above, describe the phylogenetic patterns of $\beta$-diversity for bacterial communities in the Indiana ponds.   

> ***Answer 6***: Something was going wrong with my PERANOVA but the constrained ordination doesn't seem to give significant results at tha $\alpha = 0.05$ level (if I'm interpreting that correctly). The sites differ from one another in terms of manty environmental variables but none seem to really be powerful discriminators.

## 6) SPATIAL PHYLOGENETIC COMMUNITY ECOLOGY

### A. Phylogenetic Distance-Decay (PDD)

A distance decay (DD) relationship reflects the spatial autocorrelation of community similarity.
That is, communities located near one another should be more similar to one another in taxonomic composition than distant communities.
(This is analagous to the isolation by distance (IBD) pattern that is commonly found when examining genetic similarity of a populations as a function of space.)
Historically, the two most common explanations for the taxonomic DD are that it reflects spatially autocorrelated environmental variables and the influence of dispersal limitation. 
However, if phylogenetic diversity is also spatially autocorrelated, then evolutionary history may also explain some of the taxonomic DD pattern.
Here, we will construct the phylogenetic distance-decay (PDD) relationship

First, calculate distances for geographic data, taxonomic data, and phylogenetic data among all unique pair-wise combinations of ponds.

In the R code chunk below, do the following:  
1. calculate the geographic distances among ponds,  
2. calculate the taxonomic similarity among ponds,  
3. calculate the phylogenetic similarity among ponds, and  
4. create a dataframe that includes all of the above information. 

```{r}


```

Now, let's plot the DD relationships:  
In the R code chunk below, do the following:  
1. plot the taxonomic distance decay relationship,   
2. plot the phylogenetic distance decay relationship, and  
3. add trend lines to each.

```{r}


```

In the R code chunk below, test if the trend lines in the above distance decay relationships are different from one another.

```{r}


```

***Question 7***: Interpret the slopes from the taxonomic and phylogenetic DD relationships. If there are differences, hypothesize why this might be.   

> ***Answer 7***: 


## SYNTHESIS

Ignoring technical or methodological constraints, discuss how phylogenetic information could be useful in your own research. 
Specifically, what kinds of phylogenetic data would you need? How could you use it to answer important questions in your field? 
In your response, feel free to consider not only phylogenetic approaches related to phylogenetic community ecology, but also those we discussed last week in the PhyloTraits module, or any other concepts that we have not covered in this course.

>  In my own research, I almost require a phylogeny. I use concatenated ribosomal genes that *in principle* will make a nice well resolved phylogeny of bacterial species with and without sporulation genes. This will allow me to look at evolutionary patterns of sporulation loss and potentially even help me Identify different sets of sporulation genes corresponding to different shared evolutionary histories. I want to use my sporulation predictor to look at the kinds of communities in which sporulation may be more likely to be lost and I could use the phylogeny to look at dispersion in those communities. 